# Deep Learning in Spatial Transcriptomics Analysis
Our goal is to help the scientific community by providing a (continously updated) list of various deep learning models used in the various stages of spatial transcriptomics analysis. 

We would really appreciate your contributions, so please do not hesitate to do a PR with data on a new paper and/or tool.

## Methods

| Stage/Category                             | Model (with GitHub Link)                                                | Title of Paper                                                                                                | Language            | Year | Reference                                                                                                                                                                                                                                             | Additional Notes |
|------------------------------------|-------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------------------|---------------------|------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|------------------|
| ***Spatial Reconstruction***           | [DEEPsc](https://github.com/fmaseda/DEEPsc)                             | DEEPsc: A Deep Learning-Based Map Connecting Single-Cell Transcriptomics and Spatial Imaging Data             | MATLAB              | 2021 | Maseda F, Cang Z, Nie Q. 2021. DEEPsc: A Deep Learning-Based Map  Connecting Single-Cell Transcriptomics and Spatial Imaging Data. Front  Genet 12: 636743.                                                                                           |                  |
| ***Spatial Reconstruction***           | [HematoFatePrediction](https://github.com/marrlab/HematoFatePrediction) | Prospective identification of hematopoietic lineage choice by deep learning                                   | MATLAB              | 2017 | Buggenthin F, Buettner F, Hoppe PS, Endele M, Kroiss M, Strasser M,  Schwarzfischer M, Loeffler D, Kokkaliaris KD, Hilsenbeck O, et al. 2017.  Prospective identification of hematopoietic lineage choice by deep  learning. Nat Methods 14: 403–406. |                  |
|                                    |                                                                         |                                                                                                               |                     |      |                                                                                                                                                                                                                                                       |                  |
| ***Data Integration (scRNAseq + ST)*** | [Tangram](https://github.com/broadinstitute/Tangram)                    | Deep Learning and Alignment of Spatially Resolved Single-Cell Transcriptomes with Tangram                     | Python (PyTorch)    | 2021 | Biancalani, T., Scalia, G., Buffoni, L. et al. Deep learning and alignment of spatially resolved single-cell transcriptomes with Tangram.Nat Methods 18, 1352–1362 (2021). https://doi.org/10.1038/s41592-021-01264-7.                                |                  |
| ***Data Integration (scRNAseq + ST)*** | [ST-Net](https://github.com/bryanhe/ST-Net)                             | Integrating Spatial Gene Expression and Breast Tumour Morphology via Deep Learning                            | Python (PyTorch)    | 2021 | He B, Bergenstråhle L, Stenbeck L, Abid A, Andersson A, Borg Å, Maaskola  J, Lundeberg J, Zou J. 2020. Integrating spatial gene expression and  breast tumour morphology via deep learning. Nat Biomed Eng 4: 827–834.                                |                  |
|                                    |                                                                         |                                                                                                               |                     |      |                                                                                                                                                                                                                                                       |                  |
| ***Integrative Tookit***               | [GLUER](https://github.com/software-github/GLUER)                       | GLUER: integrative analysis of multi-omics and imaging data at single-cell resolution by deep neural networks | Python (TensorFlow) | 2021 | GLUER: integrative analysis of single-cell omics and imaging data by deep neural network. Tao Peng, Gregory M. Chen, KaiTan. bioRxiv 2021.01.25.427845; doi: https://doi.org/10.1101/2021.01.25.427845                                                |                  |
|                                    |                                                                         |                                                                                                               |                     |      |                                                                                                                                                                                                                                                       |                  |




## Review Papers

A review paper on the applications of deep learning in single-cell omics analysis can be found [here](https://www.biorxiv.org/content/10.1101/2021.11.26.470166v2).
A review on deep learning in spatial transcriptomics can be found [here]() [coming very soon!].


